speechbrain/utils/recipe_tests.py

```diff
@@ -0,0 +1,453 @@
+        File where standard output is stored.
+    stderr_file: path
+        File where standard error is stored.
+
+    Returns
+    ---------
+    rc: bool
+        The return code obtained after running the command. If 0, the test is
+        run without errors. If >0 the execution failed.
+    """
+    try:
+        os.makedirs(folder)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+
+def run_recipe_tests(
+    recipe_csvfile="tests/recipes.csv",
+    script_field="Script_file",
+    hparam_field="Hparam_file",
+    recipe_id_field="RecipeID",
+    test_field="test_debug_flags",
+    check_field="test_debug_checks",
+    run_opts="--device=cpu",
+    output_folder="tests/recipe_tests/",
+    filters_fields=[],
+    filters=[],
+    do_checks=True,
+):
+    """Runs the recipes tests.
+
+    Arguments
+    ---------
+    recipe_csvfile: path
+        Path of the csv recipe file summarizing all the recipes in the repo.
+    script_field: str
+        Field of the csv recipe file containing the path of the script to run.
+    hparam_field: str
+        Field of the csv recipe file containing the path of the hparam file.
+    recipe_id_field: str
+        Field of the csv recipe file containing the unique recipe ID.
+    test_field: string
+        Field of the csv recipe file containing the test flags.
+    check_field: string
+        Field of the csv recipe file containing the checks to perform.
+    run_opts: string
+        Additional flags to add for the tests (see run_opts of speechbrain/core.py).
+    output_folder: string
+        Folder where the output of the tests are saved.
+    filter_fields: list
+        This can be used with the "filter" variable
+        to run only some tests. For instance, filter_fileds=['Task'] and filters=['ASR'])
+        will only run tests for ASR recipes.
+    filters: list
+        See above.
+    do_checks:
+        If True performs the checks on the output folder (when the check_field is not empty).
+
+    Returns
+    ---------
+    check: True
+        True if all the recipe tests pass, False otherwise.
+
+    """
+    # Create the output folder (where the tests results will be saved)
+    create_folder(output_folder)
+    print("Test ouputs will be put in %s" % (output_folder))
+
+    # Read the csv recipe file and detect which tests we have to run
+    test_script, test_hparam, test_flag, test_check = prepare_test(
+        recipe_csvfile,
+        script_field,
+        hparam_field,
+        filters_fields=filters_fields,
+        filters=filters,
+    )
+
+    # Run  script (check how to get std out, std err and save them in files)
+    check = True
+    for i, recipe_id in enumerate(test_script.keys()):
+        print(
+            "(%i/%i) Running test for %s..."
+            % (i + 1, len(test_script.keys()), recipe_id)
+        )
+
+        output_fold = os.path.join(
+            output_folder, recipe_id
+        )  # Remove folder from the last run
```

Comment:
Does this code actually remove folder from the last run? It doesn't look like it does to me.

Reply:
No, this is something we can avoid as it is not really necessary.